datalab/datalab_session/data_operations/rgb_stack.py

```diff
@@ -59,30 +59,28 @@ def wizard_description():
     def operate(self):
         rgb_input_list = self.input_data['red_input'] + self.input_data['green_input'] + self.input_data['blue_input']
 
-        if len(rgb_input_list) == 3:
-            log.info(f'Executing RGB Stack operation on files: {rgb_input_list}')
+        if len(rgb_input_list) != 3:
+            raise ClientAlertException('RGB stack requires exactly 3 files')
+        
+        log.info(f'Executing RGB Stack operation on files: {rgb_input_list}')
 
-            fits_paths = []
-            for file in rgb_input_list:
-                fits_paths.append(get_fits(file.get('basename')))
-                self.set_percent_completion(self.get_percent_completion() + 0.2)
-            
-            large_jpg_path, small_jpg_path = create_jpgs(self.cache_key, fits_paths, color=True)
+        fits_paths = []
+        for file in rgb_input_list:
+            fits_paths.append(get_fits(file.get('basename')))
+            self.set_operation_progress(self.get_operation_progress() + 0.2)
```

Comment:
This is a generic comment and I know we had already been doing this before, but calling the `.get_operation_progress()` every time we want to set the value causes an extra lookup into the remote cache, which is wasteful. We should probably just maintain a local variable with the operation progress in the class and use this during the operate method so we don't waste time retrieving it from the cache each time we set it.